common/src/objects/state.rs

```diff
@@ -86,10 +86,11 @@ impl State {
             ));
         }
 
-        let re = Regex::new(r"^[a-zA-Z0-9_-]+[a-zA-Z0-9_-]*$").unwrap();
+        let re_workloads = Regex::new(r"^[a-zA-Z0-9_-]+[a-zA-Z0-9_-]*$").unwrap();
```

Comment:
Why not just specifying r"^[a-zA-Z0-9_-]+$"?

Reply:
Can the workloads name also be just ""?

Reply:
As agreed, "^[a-zA-Z0-9_-]+$" is the minimal regex to match, `workload-Version_1`, but not `workload.Version.1`.